src/qga.rs

```diff
@@ -45,6 +45,32 @@ impl QgaWrapper {
     /// `sock` is the path to the QGA socket.
     /// `has_kvm` whether or not host supports KVM
     pub fn new(sock: PathBuf, has_kvm: bool) -> Result<Self> {
+        let timeout = if has_kvm {
+            KVM_TIMEOUT
+        } else {
+            EMULATE_TIMEOUT
+        };
+
+        // If we try reading the socket too  early, we'll hang forever and never run the test.
+        // So do the guest_sync first with a timeout to ensure that the VM Guest Agent is up.
+        let end = Instant::now() + timeout;
+        while Instant::now() < end {
+            let qga_stream = match UnixStream::connect(&sock) {
+                Ok(s) => s,
+                Err(e) => {
+                    // If we fail to connect to socket, exit this thread. The main
+                    // thread will detect a hangup and error accordingly.
+                    error!("Failed to connect QGA: {}", e);
+                    return Err(e.into());
```

Comment:
I'm wondering if hard erroring on failure here is causing the CI failure. Maybe we want to only hard error if it's the final try before timeout

Reply:
Maybe...tests pass locally for me.

Reply:
In the two failures, the VM doesn't actually seem to reach linux based on logs. wonder why

Reply:
Oh, I was suggesting there might be a race here. Previous we did a bit more work before the guest-sync. Now we do it early. It's possible we are racing?

Yeah also unclear why boot logs not coming out

Reply:
The PR I just opened had its CI succeed first try. I am suspecting something in this PR is causing a new issue

Reply:
I can repro this issue on master...somehow. I think these tests are flaky. Will try to investigate deeper...

Reply:
nvm, I'm just dumb. Forgot that the tests need qemu-ga installed on the host :-/